exploration/0002-expression-attributes.md

```diff
@@ -38,18 +38,59 @@ At least the following expression attributes should be considered:
 
   - `fallback` — A value to use instead of the default fallback,
     should the expression's primary formatting fail in some way.
+
+    > Example that would format as "Hello world" if `$place` is not defined:
+    >
+    > ```
+    > {Hello {$place @fallback=world}}
+    > ```
+
   - `locale` — An override for the locale used to format the expression.
     Should be expressed as a non-empty sequence of BCP 47 language codes.
+
+    > Example embedding a French literal in an English message:
+    >
+    > ```
+    > {In French, "{|bonjour| @locale=fr}" is a greeting}
+    > ```
+
   - `dir` — An override for the LTR/RTL/auto directionality of the expression.
 
+    > Example explicitly isolating the directionality of a placeholder:
+    >
+    > ```
+    > {Welcome, {$username @dir=auto}}
+    > ```
+
 - Attributes relevant for translators, tools, and other message operations,
   but with no runtime impact:
 
   - `example` — A literal value representing
     what the expression's formatted value will look like.
+
+    > Example providing a translator with an example for `$place`:
+    >
+    > ```
+    > {Hello {$place @example=world}}
```

Comment:
This is treading the semantic comments territory. Perhaps it would be best to leave this out of scope?

Reply:
With semantic comments we may reasonably describe values such as `$place`, but not specific expressions, which may even be repeated in variant patterns. So expression attributes such as `@example` and `@note` should be considered.